docs/fundamentals/networking/sockets/tcp-classes.md

```diff
@@ -78,10 +78,203 @@ The preceding C# code:
+socket.Bind(ep);
+```
+
+Another `TcpClient` constructor overload accepts a `hostname` and `port` as parameters, the difference from the default constructor is that this constructor tries to connect to the given `hostname` and `port`. Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient("www.example.com", 80);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+socket.Connect("www.example.com", 80);
+```
+
+### Connect to server
+
+Basically, every `Connect`, `ConnectAsync`, `BeginConnect` and `EndConnect` overload in `TcpClient` are functionally equivalent and their function signatures are same as well. Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient();
+client.Connect("www.example.com", 80);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+socket.Connect("www.example.com", 80);
+```
+
+### Create a server
+
+In `TcpListener` class, there are two constructors. Basically, both of them are initializes underlying socket. One of them is `TcpListener(IPAddress localaddr, int port)`. Consider the following TCP listener code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+```
+
+Other one is `TcpListener(IPEndPoint localEP)`, which does the same thing with the above constructor. Consider the following TCP listener code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var listener = new TcpListener(ep);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+```
+
+Also, there is a `static` <xref:System.Net.Sockets.TcpListener.Create%2A> method as well. `Create` method creates _dual-stack_ socket if [IPv6 is supported (Socket.OSSupportsIPv6 == true)](<xref:System.Net.Sockets.Socket.OSSupportsIPv6>), otherwise creates IPv4 socket. Consider the following TCP listener code:
+
+```csharp
+var listener = TcpListener.Create(5000);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(Socket.OSSupportsIPv6 ? IPAddress.IPv6Any : IPAddress.Any, 5000);
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+```
+
+### Start listening on server
+
+In `Socket` class there are two steps to start listening on a server, but in `TcpClient` calling `Start` function is enough. Consider the following TCP listener code:
```

Comment:
```suggestion
The <xref:System.Net.Sockets.TcpClient.Start> method is a wrapper combining Socket's <xref:System.Net.Socket.Bind> and <xref:System.Net.Socket.Listen>.

Consider the following TCP listener code:
```
(Haven't checked the links in my suggestion, and the result may need some cosmetics.)